lib/rspec/mocks/message_expectation.rb

```diff
@@ -683,6 +721,22 @@ def call(*_args_to_ignore, &_block)
       end
     end
 
+    # Handles the implementation of an `and_invoke` implementation.
+    # @private
+    class AndInvokeImplementation
+      def initialize(procs_to_invoke)
+        @procs_to_invoke = procs_to_invoke
+      end
+
+      def call(*_args_to_ignore, &_block)
```

Comment:
The arguments should probably be passed to the procs similar to how blocks are called...

Reply:
The original intent was to chain returns and rescues, but I don't see why this can't share the same block semantics, will fix.

Reply:
Chain returns is easy with `and_return(1,2,3,4)` etc, so this is a custom implementation so should respect the same rules, e.g. `and_return { |args| expect(args).to eq [] }` is valid and used to validate arguments, so its surprising if this doesn't also have access to args, as its essentially the "chain" form of `and_return { ... }`

Reply:
Agreed, implemented in latest commit. :+1: